components/eamxx/src/share/io/scream_output_manager.cpp

```diff
@@ -162,7 +162,7 @@ setup (const ekat::Comm& io_comm, const ekat::ParameterList& params,
       // NOTE: if you change the output frequency when you restart, this could lead to wonky behavior
       m_output_control.nsamples_since_last_write = m_run_t0.get_num_steps() % m_output_control.frequency; 
 
-      if (has_restart_data) {
+      if (has_restart_data && m_output_control.nsamples_since_last_write>0) {
```

Comment:
Actually, I think line 163 isn't correct either.  If frequency was 1 / month then `get_num_steps() % m_output_control.frequency==0` for all steps because frequency is 1, even for a run that restarts after 1 week.

Reply:
Yes, that's true. It is implicitly assumed that the number of samples is always the number of steps. Imho, this points to the need of distinguishing the sample size from the output frequency.

Reply:
Actually, so long as we use all timesteps when performing averages, the two are the same.